src/main/java/io/github/novacrypto/base58/Base58EncoderDecoder.java

```diff
@@ -23,36 +23,50 @@
 
 import java.util.Arrays;
 
-import static io.github.novacrypto.base58.CapacityEstimator.estimateMaxLength;
+import static io.github.novacrypto.base58.CapacityCalculator.maximumBase58StringLength;
 
 final class Base58EncoderDecoder implements GeneralEncoderDecoder {
 
     private static final char[] DIGITS = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz".toCharArray();
     private static final int[] VALUES = initValues(DIGITS);
 
     private final WorkingBuffer workingBuffer;
+    private final StringBuilderEncodeTarget target = new StringBuilderEncodeTarget();
 
     Base58EncoderDecoder(final WorkingBuffer workingBuffer) {
         this.workingBuffer = workingBuffer;
     }
 
     @Override
     public String encode(final byte[] bytes) {
-        final StringBuilderEncodeTarget target = new StringBuilderEncodeTarget(estimateMaxLength(bytes.length));
-        encode(bytes, target);
+        target.clear();
+        encode(bytes, target, target);
```

Comment:
As it already shares a working buffer, this object is not threadsafe and might as well share an encode target and string builder.